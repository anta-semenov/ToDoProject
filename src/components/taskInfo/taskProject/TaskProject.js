import PropTypes from 'prop-types'
import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import './TaskProject.less'

const TaskProject = ({ taskProject, projects, onProjectChange }) => (
  <div className="projects">
    <div className="projects__title">Project</div>
    {projects && projects.size > 0 ? (
      <select
        className={`projects__selector${taskProject ? ' is-selected' : ''}`}
        value={taskProject || ''}
        onChange={e => onProjectChange(e.target.options.item(e.target.selectedIndex).value)}
      >
        <option value="">No Project</option>
        {projects.toArray().map(project => (
          <option key={project.get('id')} value={project.get('id')}>
            {project.get('title')}
          </option>
        ))}
      </select>
    ) : (
      <div className={'projects__empty-list'}>You haven't create any projects yet.</div>
    )}
  </div>
)

TaskProject.propTypes = {
  taskProject: PropTypes.string,
  projects: ImmutablePropTypes.mapOf(
    ImmutablePropTypes.contains({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      completed: PropTypes.bool.isRequired
    })
  ),
  onProjectChange: PropTypes.func.isRequired
}

export default TaskProject
