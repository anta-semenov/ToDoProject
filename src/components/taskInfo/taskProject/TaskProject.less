@import '../../../styles/variables.less';

.projects {
  position: relative;
  margin: 20px 0 30px 8px;
}
.projects__title {
  position: relative;
  margin: 0;
  font-size: 10px;
  font-weight: 400;
  text-transform: uppercase;
  color: @color-font-light;
}
.projects__selector {
  appearance: none;
  position: relative;
  width: 100%;
  padding: 5px 6px 4px 8px;
  margin: 4px 0 0 0;
  cursor: pointer;
  transition: border-color @transition-basic, color @transition-basic, background-color @transition-basic;

  border: 1px solid @color-border-gray;
  border-radius: 2px;
  font-size: @font-size-small;
  color: @color-font-light;
  font-family: 'Fira Sans', sans-serif;

  background-image: url("data:image/svg+xml;charset=UTF-8, <svg viewBox='0 0 12 6' xmlns='http://www.w3.org/2000/svg'><polygon points='0 0.5 5.58044434 6 11.1350098 0.5 10.5786566 0.00178777393 5.58044434 5 0.580444336 0' fill='@{color-font-light}'></polygon></svg>");
  background-position: 97% center;
  background-size: 6%;
  background-repeat: no-repeat;

  &:hover {
    border-color: @color-border-dark-gray;
    color: @color-font-medium;
    background-image: url("data:image/svg+xml;charset=UTF-8, <svg viewBox='0 0 12 6' xmlns='http://www.w3.org/2000/svg'><polygon points='0 0.5 5.58044434 6 11.1350098 0.5 10.5786566 0.00178777393 5.58044434 5 0.580444336 0' fill='@{color-font-medium}'></polygon></svg>");
  }
  &:active,
  &:focus {
    border-color: @color-border-dark-gray;
    background-color: @color-bg-light-gray;
    outline: none;
    color: @color-font-medium;
    background-image: url("data:image/svg+xml;charset=UTF-8, <svg viewBox='0 0 12 6' xmlns='http://www.w3.org/2000/svg'><polygon points='0 0.5 5.58044434 6 11.1350098 0.5 10.5786566 0.00178777393 5.58044434 5 0.580444336 0' fill='@{color-font-medium}'></polygon></svg>");
  }
  &.is-active {
    background-color: @color-accent;
    border-color: @color-accent;
    color: #ffffff;

    &:hover {
      background-color: darken(@color-accent, 10%);
      border-color: darken(@color-accent, 10%);
    }
  }
}
.projects__empty-list {
  position: relative;
  font-size: @font-size-basic;
  font-weight: @font-weight-light;
}
