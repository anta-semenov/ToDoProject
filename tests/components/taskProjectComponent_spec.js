import React from 'react'
import { mount } from 'enzyme'
import { fromJS } from 'immutable'
import TaskProject from '../../src/components/taskInfo/taskProject/TaskProject.js'

describe('TaskProject Component', () => {
  it('Should render component without projects', () => {
    const taskProject1 = mount(<TaskProject onProjectChange={() => {}} />)
    expect(taskProject1).toMatchSnapshot()
    const taskProject2 = mount(<TaskProject projects={fromJS({})} onProjectChange={() => {}} />)
    expect(taskProject2).toMatchSnapshot()
  })
  it('Should render component with projects, but without selected project', () => {
    const projects = fromJS({
      bh52ogy5s0fm: {
        id: 'bh52ogy5s0fm',
        title: 'Test project 0',
        completed: false
      },
      bh52ogy5s0f1: {
        id: 'bh52ogy5s0f1',
        title: 'Test project 1',
        completed: false
      }
    })
    const taskProject = mount(<TaskProject projects={projects} onProjectChange={() => {}} />)
    expect(taskProject).toMatchSnapshot()
  })
  it('Should render component with selected project', () => {
    const projects = fromJS({
      bh52ogy5s0fm: {
        id: 'bh52ogy5s0fm',
        title: 'Test project 0',
        completed: false
      },
      bh52ogy5s0f1: {
        id: 'bh52ogy5s0f1',
        title: 'Test project 1',
        completed: false
      }
    })
    const taskProject = mount(
      <TaskProject taskProject="bh52ogy5s0fm" projects={projects} onProjectChange={() => {}} />
    )
    expect(taskProject).toMatchSnapshot()
  })
})
