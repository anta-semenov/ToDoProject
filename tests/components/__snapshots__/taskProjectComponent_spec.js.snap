// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskProject Component Should render component with projects, but without selected project 1`] = `
<TaskProject
  onProjectChange={[Function]}
  projects={
    Immutable.Map {
      "bh52ogy5s0fm": Immutable.Map {
        "id": "bh52ogy5s0fm",
        "title": "Test project 0",
        "completed": false,
      },
      "bh52ogy5s0f1": Immutable.Map {
        "id": "bh52ogy5s0f1",
        "title": "Test project 1",
        "completed": false,
      },
    }
  }
>
  <div
    className="projects"
  >
    <div
      className="projects__title"
    >
      Project
    </div>
    <select
      className="projects__selector"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        No Project
      </option>
      <option
        key="bh52ogy5s0fm"
        value="bh52ogy5s0fm"
      >
        Test project 0
      </option>
      <option
        key="bh52ogy5s0f1"
        value="bh52ogy5s0f1"
      >
        Test project 1
      </option>
    </select>
  </div>
</TaskProject>
`;

exports[`TaskProject Component Should render component with selected project 1`] = `
<TaskProject
  onProjectChange={[Function]}
  projects={
    Immutable.Map {
      "bh52ogy5s0fm": Immutable.Map {
        "id": "bh52ogy5s0fm",
        "title": "Test project 0",
        "completed": false,
      },
      "bh52ogy5s0f1": Immutable.Map {
        "id": "bh52ogy5s0f1",
        "title": "Test project 1",
        "completed": false,
      },
    }
  }
  taskProject="bh52ogy5s0fm"
>
  <div
    className="projects"
  >
    <div
      className="projects__title"
    >
      Project
    </div>
    <select
      className="projects__selector is-selected"
      onChange={[Function]}
      value="bh52ogy5s0fm"
    >
      <option
        value=""
      >
        No Project
      </option>
      <option
        key="bh52ogy5s0fm"
        value="bh52ogy5s0fm"
      >
        Test project 0
      </option>
      <option
        key="bh52ogy5s0f1"
        value="bh52ogy5s0f1"
      >
        Test project 1
      </option>
    </select>
  </div>
</TaskProject>
`;

exports[`TaskProject Component Should render component without projects 1`] = `
<TaskProject
  onProjectChange={[Function]}
>
  <div
    className="projects"
  >
    <div
      className="projects__title"
    >
      Project
    </div>
    <div
      className="projects__empty-list"
    >
      You haven't create any projects yet.
    </div>
  </div>
</TaskProject>
`;

exports[`TaskProject Component Should render component without projects 2`] = `
<TaskProject
  onProjectChange={[Function]}
  projects={Immutable.Map {}}
>
  <div
    className="projects"
  >
    <div
      className="projects__title"
    >
      Project
    </div>
    <div
      className="projects__empty-list"
    >
      You haven't create any projects yet.
    </div>
  </div>
</TaskProject>
`;
